### ISSUE TITLE
Support for missing/NaN values in multi-target labels for multi-output regression (multi_output_tree)

### ISSUE THREAD

**Issue by WXJT-79** [dmlc/xgboost#12295]
We're training a multi-target regression model (18 correlated targets) where a substantial fraction of samples have some, but not all, targets observed. The unobserved targets aren't "no signal" or zero; they're simply not measured for that sample. We'd like the model to learn from the observed targets per-sample without requiring us to impute a value for the unobserved ones, similar to how single-output XGBRegressor handles this naturally (fit 18 independent models, each only trained on rows where its target is present).
Because multi_output_tree shares tree structure across targets (which is the specific benefit we want, learning correlations between targets), we can't just fall back to per-target independent models without losing that benefit.

Is per-target missing-label masking (i.e., excluding specific (sample, target) pairs from the gradient/Hessian computation rather than imputing them) on the roadmap for multi_output_tree?

Environment:
XGBoost version: 3.2.0
Python
tree_method="hist", multi_strategy="multi_output_tree"

**Comment by trivialfis** [comment]
Not at the moment. We don't have a good algorithm for dealing with missing labels, which is not the same thing as missing features. Maybe XGBoost can just omit those gradients, but I don't know the effect. You can test by defining a custom objective where missing labels return 0 as gradient and Hessian.
